cylc/uiserver/websockets/template.py

```diff
@@ -0,0 +1,130 @@
+# This file is a temporary solution for subscriptions with graphql_ws and
+# Tornado, from the following pending PR to graphql-ws:
+# https://github.com/graphql-python/graphql-ws/pull/25/files
+# The file was copied from this revision:
+# https://github.com/graphql-python/graphql-ws/blob/cf560b9a5d18d4a3908dc2cfe2199766cc988fef/examples/tornado/template.py
+
+from string import Template
+
+
+def render_graphiql(base_url: str) -> str:
```

Comment:
From the PR's example.

Reply:
Modified to create the WebSocket subscription URL using JS for hostname and port, and Python for the base URL.